### DISCUSSION TITLE
Can you fingerprint out whose device it is regardless of browser in a mobile environment?

### DISCUSSION THREAD

**Discussion by DongJu-Na** [fingerprintjs/fingerprintjs#886] [General]
Hello, I'm a junior developer working on fingerprints.

Is there a way to specify whose device fingerprint technology is in a mobile environment? Can you tell me if you know?

Let me give you an example.

Here are two friends, myself included
Let's assume that all cell phone models are the same.

  
|browser|me|A|B|
|------|---|---|---|
|chrome|key1|key2|key3|
|edge|key1|key2|key3|
|safari|key1|key2|key3|

I'm asking because I think github here is the most active.

Thank you.

**Comment by makma** [comment]
Hey @DongJu-Na ,
This is rather a research task than a question. Moreover, you are confusing browser and device fingerprinting (based on your `keyX` for the specific person among multiple browsers on the same device). This question is beyond the scope of FingerprintJS repository or browser fingerprinting. Commercial and closed source Fingerprint Pro can do that (but only for the native iOS and Android contexts, not for the web JS context). You can take a look at some naive native open-source techniques in our repositories for [iOS](https://github.com/fingerprintjs/fingerprintjs-ios) and [Android](https://github.com/fingerprintjs/fingerprintjs-android).

**Reply by DongJu-Na** [reply]
First of all, thank you for moving the post. Can I understand that it is a technology that cannot be implemented with JavaScript?

**Reply by DongJu-Na** [reply]
Isn't that possible in browser javascript?

**Reply by makma** [reply]
> Can I understand that it is a technology that cannot be implemented with JavaScript?
> Isn't that possible in browser javascript?

No, it would be bold from me to say that it's not possible, however, we don't do that and I'm not aware of any approach.

**Comment by DongJu-Na** [comment]
Thank you for your reply.
If you want to solve a problem like me or have solved it, please leave a comment
Thank you. 😥
